### DISCUSSION TITLE
Dashboard unable to show graph even though NBXplorer running on postgres

### DISCUSSION THREAD

**Discussion by Gusep** [btcpayserver/btcpayserver#5292] [Technical Support] [answered]
Hi, I have a manual install of BTCPayserver  v1.11.2. 

My dashboard is still showing the alert:

_We would like to show you a chart of your balance. Please migrate to the new NBXplorer backend for that data to become available._

But all looks like NBXplorer is using postgres backend succesfully. 

**What might be the problem or how could I debug it deeper?**

NBXplorer config file:
```
btc.rpc.auth=bitcoin:somepassword
port=24445
mainnet=1
postgres=User ID=someuser;Password=somepassword;Host=localhost;Port=5432;Database=nbxplorer;
automigrate=1
nomigrateevts=1
```
NBXplorer daemon is running and showing the info:

```
info: NBXplorer.DatabaseSetup: Postgres services activated
```

BTCPayServer config file looks like this:
```
### Global settings ###
network=mainnet
chains=btc
debuglog=/home/bitcoin/.btcpayserver/debug.log
debugloglevel=debug

### Server settings ###
port=23001
bind=0.0.0.0

### Database ###
postgres=User ID=someuser;Password=somepassword;Host=localhost;Port=5432;Database=btcpayserver;

### NBXplorer settings ###
BTC.explorer.url=http://127.0.0.1:24445/
BTC.explorer.cookiefile=/home/bitcoin/.nbxplorer/Main/.cookie
BTC.blockexplorerlink=https://mempool.space/tx/{0}
BTC.lightning=type=lnd-rest;server=https://127.0.0.1:8080/;macaroonfilepath=/home/bitcoin/.lnd/data/chain/bitcoin/mainnet/admin.macaroon;certthumbprint=8B2A1D0217C059F423F2CF2C681211C904153908FGS005D039C7A9511A01A21A
```

**Comment by Gusep** [comment]
Anyone pls?

**Comment by Kixunil** [comment, marked as answer]
You need to add this line to btcpay config:

```
explorer.postgres=User ID=someuser;Password=somepassword;Host=localhost;Port=5432;Database=nbxplorer;
```

(Same user/password as the one used for nbxplorer)

**Reply by Gusep** [reply]
That did the trick, thanks!

**Comment by kosuodhmwa** [comment]
how to use on command line?

BTW: "User ID" seems to be wrong - at least in my case. I need to configure DB settings (via cmd line) as this:

./run.sh --port 8080 --bind 0.0.0.0 --postgres "Username=postgres;password=MY_PASS;Host=127.0.0.1;Port=5432;Database=btcpayserver";

So how to add that "explorer.postgres=" parameter here?

thx in advance for feedback(s)

**Comment by kosuodhmwa** [comment]
same issue ("User ID=" vs. "Username=") on nbx: https://github.com/dgarage/NBXplorer/issues/498 

Maybe somebody knows why (all official manuals write about "User ID=" and not about "Username="...!?)

**Comment by kosuodhmwa** [comment]
When i start it like that: 

./run.sh --port 8080 --bind 0.0.0.0 --explorer.postgres "Username=postgres;password=XXX;Host=127.0.0.1;Port=5432;Database=btcpayserver" --postgres "Username=postgres;password=XXX;Host=127.0.0.1;Port=5432;Database=btcpayserver";


i get that message here

![image](https://github.com/user-attachments/assets/daebf21c-050f-4e33-ab70-b738d299a73d)

**Comment by kosuodhmwa** [comment]
only with '--postgres' param, i had that problem from beginning, from that first usage of installation using postgres

https://github.com/btcpayserver/btcpayserver/issues/6486

**Comment by kosuodhmwa** [comment]
this 
![image](https://github.com/user-attachments/assets/e763bdfc-ec47-4ca9-a4a0-f904f36139f6)

**Comment by kosuodhmwa** [comment]
or is it just cause there are 0 transactions yet?

**Comment by kosuodhmwa** [comment]
NBXplorer version is **2.5.16**

**Comment by kosuodhmwa** [comment]
![image](https://github.com/user-attachments/assets/81d15a0d-5ce5-4469-bab8-516da8bb3d76)
**--deleteaftermigration** param does also cause a problem ... 

-> https://github.com/dgarage/NBXplorer/blob/master/docs/Postgres-Migration.md
